python/setup.py

```diff
@@ -250,14 +257,22 @@ def run(self):
         license='http://www.apache.org/licenses/LICENSE-2.0',
         # Don't forget to update python/docs/source/getting_started/install.rst
         # if you're updating the versions or dependencies.
-        install_requires=['py4j==0.10.9.2'],
+        install_requires=[
+            'py4j==0.10.9.2',
+            'pandas>=0.23.2',
+            'pyarrow>=0.10',
+            'numpy>=1.14,<1.20.0',
+        ],
         extras_require={
             'ml': ['numpy>=1.7'],
             'mllib': ['numpy>=1.7'],
             'sql': [
                 'pandas>=%s' % _minimum_pandas_version,
                 'pyarrow>=%s' % _minimum_pyarrow_version,
```

Comment:
nit: pyarrow minimun version is introduced in _minimum_pyarrow_version, but if we introduce the `pyarrow>=0.10` for pandas_on_spark in here, maybe we could change the `_minimum_pyarrow_version` to something like `_minimum_sql_pyarrow_version`

[1] https://github.com/apache/spark/pull/32386/files#diff-eb8b42d9346d0a5d371facf21a8bfa2d16fb49e213ae7c21f03863accebe0fcfR115

Reply:
Good catch! There is only one usage of `_minimum_pyarrow_version`. How about removing the variable and using the value instead?

Reply:
If we can use the same lower bound, I prefer just use `_minimum_pyarrow_version` directly as suggestion in  https://github.com/apache/spark/pull/32386#discussion_r623528856

```python
'pandas_on_spark': [
    'pandas>=%s' % _minimum_pandas_version,
    'pyarrow>=%s', % _minimum_pyarrow_version,
]
```